app/models/dialog_import_validator.rb

```diff
@@ -11,7 +11,7 @@ def initialize(dialog_field_association_validator = DialogFieldAssociationValida
   end
 
   def determine_validity(import_file_upload)
-    potential_dialogs = YAML.safe_load(import_file_upload.uploaded_content, [Symbol])
+    potential_dialogs = YAML.safe_load(import_file_upload.uploaded_content, :permitted_classes => [Symbol])
```

Comment:
Do these changes fix the this code without the `YamlLoadAliases`?

So are we deciding if we want this change or the Yaml monkey patch?

Reply:
Good question.  When possible, safe_load is best.  For tests, it's ok for them to use unsafe_load if enumerating the list isn't worth it.  Ultimately, the suggestion to log warnings below will help us build the list of permitted classes that we can use in most places.

Reply:
Looking at psych code, it seems to call `to_s` on the passing in permitted classes so maybe I can try another way of doing it.  If I recall, the rails `config.active_record.yaml_column_permitted_classes` didn't allow string/symbols but maybe if I can get in the middle, I can call safe_load directly with string/symbols and avoid the issue with reload and eager loading class objects.

Reply:
I feel like there are 2 use cases:

- use safe_load when ever our code references `YAML`. I think this is a good trend
- use safe_load when rails models have a yaml column -- this one I believe should use unsafe_load via the configuration parameter

Reply:
adding to that comment

I believe when we use YAML.safe_load, the list of models is relatively small. MiqExpression and a few others for reports. Also, since this is from external parties, this should be locked down more.

I believe when we use active record yaml serialization (e.g. with MiqQueue), we want many more models. And theoretically, this is not externally accessible, so we could be more liberal with the number of models we support here. (e.g.: all?)